frontend/src/bundles/common/components/upload-video/components/dropzone/video-dropzone.tsx

```diff
@@ -0,0 +1,150 @@
+import { Input } from '@chakra-ui/react';
+import { DndContext, useDroppable } from '@dnd-kit/core';
+import { type ChangeEvent, type DragEvent } from 'react';
+
+import { Box } from '~/bundles/common/components/components.js';
+import { useCallback, useRef } from '~/bundles/common/hooks/hooks.js';
+import { notificationService } from '~/bundles/common/services/services.js';
+
+import { DropzoneInstructions } from './components/components.js';
+import {
+    DROPZONE_ID,
+    MAX_FILE_SIZE,
+    MIN_VIDEO_DURATION,
+} from './libs/constants/constants.js';
+import {
+    DOM_EVENT,
+    ERROR_ID,
+    ERROR_MESSAGE,
+    ERROR_TITLE,
+    HTML_TAG,
+} from './libs/enums/enums.js';
+import { isFileSizeValid, isVideoFile } from './libs/helpers/helpers.js';
+
+type Properties = {
+    videoSource: string | null;
+    handleRemoveVideo: () => void;
+    handleSetVideo: (fileUlr: string) => void;
+};
+
+const VideoDropzone = ({
+    videoSource,
+    handleRemoveVideo,
+    handleSetVideo,
+}: Properties): JSX.Element => {
+    const inputFile = useRef<HTMLInputElement | null>(null);
+    const { setNodeRef } = useDroppable({
+        id: DROPZONE_ID,
+    });
+
+    const handleFiles = useCallback(
+        (files: FileList): void => {
+            if (files && files.length > 0) {
+                const file = files[0];
+
+                if (!file) {
+                    return;
+                }
+
+                if (!isFileSizeValid(file, MAX_FILE_SIZE)) {
+                    notificationService.error({
+                        message: ERROR_MESSAGE.VIDEO_SIZE,
+                        id: ERROR_ID.VIDEO_SIZE,
+                        title: ERROR_TITLE.VIDEO_SIZE,
+                    });
+                    return;
+                }
+
+                if (!isVideoFile(file)) {
+                    notificationService.error({
+                        message: ERROR_MESSAGE.FILE_TYPE,
+                        id: ERROR_ID.FILE_TYPE,
+                        title: ERROR_TITLE.FILE_TYPE,
+                    });
+                    return;
+                }
+
+                const videoURL = URL.createObjectURL(file);
+
+                const videoElement = document.createElement(HTML_TAG.VIDEO);
```

Comment:
Where does this video element then go? I see that you create it here, but don't attach it to the dom or return it

Reply:
We only need this element to get metadata, like the duration, to compare it with the minimum duration.